### PR TITLE
Use the same hash fn in hashPartition and hashPartitionByKey

### PR DESCRIPTION
**Pull request by anish749**
Just noticed while reviewing the code that the hash functions used for partitioning the lhs and the rhs can have slightly different behaviors in transforms like sparseIntersectByKey. Not sure if there is a practical implication of this.

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/util/ScioUtil.scala [hunk 1]
```diff
@@ -86,4 +86,9 @@ private[scio] object ScioUtil {
 
   def pathWithShards(path: String): String =
     path.replaceAll("\\/+$", "") + "/part"
+
+  def hashCodeFn[K]: K => Int = {
```
Comment (regadas) [inline scio-core/src/main/scala/com/spotify/scio/util/ScioUtil.scala:90]: we should make this ` def hashCodeFn[K](k :K): Int`. I feel like we should change the name as well, perhaps smth like `consistentHashCode`?

File: scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala [hunk 2]
```diff
@@ -406,7 +406,7 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
   def hashPartition(numPartitions: Int): Seq[SCollection[T]] =
     self.partition(
       numPartitions,
-      t => Math.floorMod(t.hashCode(), numPartitions)
+      t => Math.floorMod(ScioUtil.hashCodeFn[K](t), numPartitions)
```
Comment (regadas) [inline scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala:409]: should be `T` right?

Comment (anish749) [reply scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: yes absolutely, I am now convinced that my coding skills are bad without a compiler!!!

**Review by regadas — COMMENTED** [review]
@anish749 👍 can you run scalafix? there's an unused import now.

**Review by regadas — APPROVED** [review]
Thanks @anish749 !

**Pull request merged**